explorer/indexer/app/db.go

```diff
@@ -101,6 +103,7 @@ func getCursor(ctx context.Context, client *ent.XProviderCursorClient, chainID u
 
 func insertBlock(ctx context.Context, tx *ent.Tx, block xchain.Block) (*ent.Block, error) {
 	b, err := tx.Block.Create().
+		SetUUID(uuid.New()).
```

Comment:
Don't the blocks have something unique already? Like height?

Reply:
true we could remove the field from the schema!

Reply:
Then, let's remove what's not needed. If height is unique that should suffice.